libs/qCC_db/src/ccGenericPrimitive.cpp

```diff
@@ -66,14 +66,33 @@ const ccGenericPrimitive& ccGenericPrimitive::operator += (const ccGenericPrimit
 	unsigned newFacesCount = facesCount + prim.size();
 	bool primHasVertNorms = prim.getAssociatedCloud()->hasNormals();
 	bool primHasFaceNorms = prim.hasTriNormals();
+	bool primHasColors = prim.getAssociatedCloud()->hasColors();
+	
+	if (primHasColors && !verts->hasColors())
+		{
+		if (verts->size() > 0 && !verts->setColor(ccColor::white))
```

Comment:
Is it necessary?

Reply:
sort of... you get a warning in the log about resizing the color table for an empty cloud

``` c++
bool ccPointCloud::reserveTheRGBTable()
{
	if (m_points.capacity() == 0)
	{
		ccLog::Warning("[ccPointCloud] Calling reserveTheRGBTable with an zero capacity cloud");
	}
```
This is called down the line when calling `setColor`
without the checks it pollutes the log each time you create a primitive which has color

Reply:
Oh, I was more asking about the 'setColor' call, but now I realize it's in case we add a primitive without any color...